src/udev/udev-rules.c

```diff
@@ -1069,6 +1069,51 @@ static UdevRuleOperatorType parse_operator(const char *op) {
         return _OP_TYPE_INVALID;
 }
 
+static void check_token_delimiters(UdevRuleLine *rule_line, const char *line) {
+        assert(rule_line);
+
+        size_t n_comma = 0;
+        bool ws_before_comma = false, ws_after_comma = false;
+        const char *p;
+
+        for (p = line; !isempty(p); ++p) {
+                if (*p == ',')
+                        ++n_comma;
+                else if (strchr(WHITESPACE, *p)) {
+                        if (n_comma > 0)
+                                ws_after_comma = true;
+                        else
+                                ws_before_comma = true;
+                } else
+                        break;
+        }
+
+        if (line == rule_line->line) {
+                /* this is the first token of the rule */
+                if (n_comma > 0)
+                        log_line_warning(rule_line, "Stray leading comma.");
+        } else if (isempty(p)) {
+                /* there are no more tokens in the rule */
+                if (n_comma > 0)
+                        log_line_warning(rule_line, "Stray trailing comma.");
+        } else {
+                /* single comma is expected */
+                if (n_comma == 0)
+                        log_line_warning(rule_line, "No comma between tokens.");
+                else if (n_comma > 1)
+                        log_line_warning(rule_line, "Too many commas between tokens.");
+
+                /* whitespace after comma is expected */
+                if (n_comma > 0) {
+                        if (ws_before_comma)
+                                log_line_warning(rule_line, "Whitespace before comma.");
+                        if (!ws_after_comma)
+                                log_line_warning(rule_line, "No whitespace after comma.");
+                } else if (!ws_before_comma && !ws_after_comma)
+                        log_line_warning(rule_line, "No whitespace between tokens.");
```

Comment:
Now it seems that `No comma between tokens.` is somewhat misleading, maybe change it to `A comma between tokens is expected.`? Likewise,
`Whitespace before comma.` -> `Stray whitespace before comma.`,
`No whitespace after comma.` -> `Whitespace after comma is expected.`, 
`No whitespace between tokens.` -> `Whitespace between tokens is expected.`

@yuwata, what do you think?

Reply:
Sounds good.